### PR TITLE
Move machine/node-specific health checks into `gardenlet`'s shoot care controller

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
This PR moves the node-specific health checks from the extensions library into `gardenlet`'s shoot care controller. With the new `MachineControllerManagerDeployment` feature gate introduced in https://github.com/gardener/gardener/pull/8018, `gardenlet` can perform the health checks itself and there is no necessity anymore that all extensions perform the check.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/7594

**Special notes for your reviewer**:
`provider-local` will be adapted as part of https://github.com/gardener/gardener/pull/8019.

**Release note**:

```feature developer
Provider extensions should be adapted such that they no longer perform health checks specific to the `machine-controller-manager` deployment or the machines/nodes. In the future, `gardenlet` will take over performing these checks. Please see https://github.com/gardener/gardener/pull/8019 for an example how `provider-local` was adapted and replicate it for your provider extensions.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
@acumino raised the concern that provider extensions might assign provider-specific error codes to the condition populated to the `extensions.gardener.cloud/v1alpha1.Worker` resource. With this PR, this would no longer be possible, and we would loose all the error codes.
I'm marking this PR as draft until this is discussed/figured out.

**Comment by rfranzke** [comment]
I have adapted the PR to remove the checks where extensions might assign provider-specific error codes. For now, they will remain in the extensions library, however `gardenlet` can perform some general checks like scale-up and scale-down.

File: pkg/operation/care/checker.go [hunk 1]
```diff
@@ -636,6 +648,86 @@ func (b *HealthChecker) CheckClusterNodes(
...
+			}
+		}
+	}
+
+	// only nodes that are managed by MCM is considered
+	registeredNodes = registeredNodes - nodeNotManagedByMCM
+
+	machineList := &machinev1alpha1.MachineList{}
+	if registeredNodes != desiredMachines || readyNodes != desiredMachines {
+		if err := b.reader.List(ctx, machineList, client.InNamespace(seedNamespace)); err != nil {
```
Comment (acumino) [inline pkg/operation/care/checker.go:678]: In the old code, we were using cached client ref- https://github.com/gardener/gardener/blob/a89de8de8dabc8a28757c0313aee882c7de415f9/extensions/pkg/controller/healthcheck/worker/nodes.go#LL144C15-L144C25, but here we are using API reader, is there a specific reason for it?

File: pkg/operation/care/checker_test.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+import (
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("health", func() {
```
Comment (acumino) [inline pkg/operation/care/checker_test.go]: Nit
```suggestion
var _ = Describe("Health", func() {
```

**Review by acumino — COMMENTED** [review]
Had a combined look with @oliver-goetz.
Other than a few comments it's lgtm!

File: pkg/operation/care/checker.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/care/checker.go:678]: We are also passing a cached client here: https://github.com/gardener/gardener/blob/b3c5440744f95006264d291c2ada9e952155b931/pkg/operation/care/health.go#L297

**Review by acumino — COMMENTED** [review]
/approve

**Comment by rfranzke** [comment]
@timuthy Please add this to the `v1.73` milestone as well :)

**Comment by oliver-goetz** [comment]
/lgtm

🚀

**Pull request merged**